src/key.rs

```diff
@@ -395,11 +395,15 @@ impl PublicKey {
 
     /// Adds the keys in the provided slice together, returning the sum. Returns
     /// an error if the result would be the point at infinity, i.e. we are adding
-    /// a point to its own negation
+    /// a point to its own negation, or if the provided slice has no element in it.
     pub fn combine_keys(keys: &[&PublicKey]) -> Result<PublicKey, Error> {
```

Comment:
I am not sure whether it is a good practice to return general error type when in fact only a single error subtype may be caused by the function. I think replacing `Result<_, Error>` on `Result<_, InvalidPublicKeySum>` will do a better job (the caller, if needed, may convert the error into general `Error` crate type).

Reply:
I can't speak for what is good practice or not, but looking at the rest of the code I feel like doing what you propose would be inconsistent with it, so if that's the way we want to go it might be better to make the change everywhere at once.

Reply:
Yes, I also have seen that all functions in the lib used to return the single error type, which is a thing is not a good API, but this will certainly require distinct PR

Reply:
If all of our functions returned distinct error types it would be an extremely unergonomic API, requring downstream users to wrap all of our individual types and implementing `From` on them to convert to their own error types.